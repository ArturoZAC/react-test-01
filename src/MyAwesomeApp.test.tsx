import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { MyAwesomeApp } from "./MyAwesomeApp";

describe("MyAwesomeApp", () => {
  test("Should render first and lastname", () => {
    const { container } = render(<MyAwesomeApp />);

    const h1 = container.querySelector("h1");
    const h3 = container.querySelector("h3");

    expect(h1?.innerHTML).toContain("Fernando");
    expect(h3?.innerHTML).toContain("Herrera");
  });

  test("Should render first and lastname - screen", () => {
    render(<MyAwesomeApp />);
    // screen.debug();

    const h1 = screen.getByTestId("first-name-title");
    expect(h1.innerHTML).toContain("Fernando");
  });

  test("Should match snapsnot", () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  });

  test("Should match snapsnot", () => {
    render(<MyAwesomeApp />);
    expect(screen.getByTestId("div-app")).toMatchSnapshot();
  });
});
