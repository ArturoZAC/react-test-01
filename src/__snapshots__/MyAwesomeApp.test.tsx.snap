// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Should match snapsnot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
       
      Fernando
       
    </h1>
    <h3>
       
      Herrera
       
    </h3>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Smash, Metal Gear
    </p>
    <p>
      4
    </p>
    <h1>
      No activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","country":"Canadá"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > Should match snapsnot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    Fernando
     
  </h1>
  <h3>
     
    Herrera
     
  </h3>
  <p
    class="mi-clase-favorita"
  >
    Elden Ring, Smash, Metal Gear
  </p>
  <p>
    4
  </p>
  <h1>
    No activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC-123","country":"Canadá"}
  </p>
</div>
`;
